### ISSUE TITLE
Floccus (XBEL) is spamming the Nextcloud activity stream

### ISSUE THREAD

**Issue by marcelschmidt1337** [marcelklehr/floccus#176]
### Software versions

* Browser: Firefox 62.0.2
* Nextcloud: 13.0.6
* Nextcloud Bookmarks app: 0.12.2
* Floccus: 3.0.8

### Steps to reproduce

1. Use XBEL in Webdav adapter
2. Sync bookmarks
3. Use it for a while, add/remove bookmarks, don't change anything for a couple of days

### Expected outcome

1. Lock file activity should be excluded from Activity stream
2. XBEL file should only be changed if a new bookmark was added or removed

### Actual outcome

Nextcloud activity stream is spammed with "XBEL.lock file created/deleted.
Also the XBEL file with the bookmarks was changed a lot, although no changes were made in my browser.

**Comment by jlbprof** [comment]
>Nextcloud activity stream is spammed with "XBEL.lock file created/deleted.

What do you mean by that?

The typical process is:

1) Check for the presence of the xbel.lock file.
2) If it is there, wait and try again in awhile (another browser could be
syncing)
3) If at the end of the try period, it is still there it gives up.

Otherwise fi the file is NOT there, it will create it
When the sync process is complete it is deleted.

Is the xbel.lock file still present on your Nextcloud?

**Comment by marcelschmidt1337** [comment]
![screenshot 2018-10-01 at 14 23 01](https://user-images.githubusercontent.com/10891885/46288304-8a838f00-c585-11e8-928c-dcbdbeac108b.png)

I attached a screenshot of my activity stream. This is only the last couple of hours.
I did not change any bookmark but the lock file is still created and the XBEL file was changed for some reason. This makes the activity feature unusable for me.

**Comment by marcelklehr** [comment]
The only entries we can reasonably get rid of are the mutations XBEL file even though nothing has changed. The lock file needs to be used because nextcloud doesn't offer native locking via DAV: https://github.com/nextcloud/server/issues/1308

**Comment by marcelklehr** [comment]
The problem is that the algorithm effectively uses a diff to determine the changes as neither the browser nor the XBEL file track changes. So, we only know whether something has to be updated on the server once we have downloaded the file. But if we do need to update the file on the server we need to make sure it hasn't changed since we downloaded it. We can only do that by setting a lock *before* downloading. In any case, If you have nextcloud installed, I generally recommend using the bookmarks app, if there's nothing to be said against it.
